ui/dialog.js

```diff
@@ -412,23 +413,24 @@ return $.widget( "ui.dialog", {
 				},
 				text: false
 			})
-			.addClass( "ui-dialog-titlebar-close" )
 			.appendTo( this.uiDialogTitlebar );
-		this._on( this.uiDialogTitlebarClose, {
-			click: function( event ) {
-				event.preventDefault();
-				this.close( event );
-			}
-		});
 
-		uiDialogTitle = $( "<span>" )
+		this._addClass( this.uiDialogTitlebarClose, "ui-dialog-titlebar-close" )
+			._on( this.uiDialogTitlebarClose, {
+				click: function( event ) {
+					event.preventDefault();
+					this.close( event );
+				}
+			});
+
+		this.uiDialogTitle = $( "<span>" )
```

Comment:
ill change it back its a relic of the older api where we needed to save everything on the instance because of the giant switch.